File: frontend/src/types/kol.ts
/**
 * Type definitions for KOL (Key Opinion Leader) data.
 * Mirrors backend Pydantic models - using camelCase to match API response.
 */

export interface KOL {
  id: string;
  name: string;
  affiliation: string;
  country: string;
  city: string | null;  // Optional - not always available in Excel data
  expertiseArea: string;
  publicationsCount: number | null;
  hIndex: number | null;
  citations: number | null;
}

export interface CountryCount {
  country: string;
  count: number;
}

export interface HighestCitationsPerPublicationKOL {
  id: string;
  name: string;
  ratio: number;
  citations: number;
  publications_count: number;
}

export interface KOLStats {
  total_kols: number;
  unique_countries: number;
  total_publications: number;
  avg_h_index: number;
  top10_countries_by_kol_count: CountryCount[];
  highest_citations_per_publication_kol: HighestCitationsPerPublicationKOL;
  data_quality_issues: string[];
}

/**
 * API error response structure
 */
export interface APIError {
  detail: string;
}

File: backend/app/models/kol.py
"""KOL (Key Opinion Leader) data models."""
from typing import Optional
from pydantic import BaseModel, Field, ConfigDict


class KOL(BaseModel):
    """
    Key Opinion Leader model representing a medical/pharmaceutical expert.
    
    All numeric fields are optional to handle potential data quality issues
    in the source data (missing or invalid values).
    """
    id: str
    name: str
    affiliation: str
    country: str
    city: Optional[str] = None  # Optional - not always available in Excel data
    expertise_area: str = Field(alias="expertiseArea")
    publications_count: Optional[int] = Field(alias="publicationsCount", default=None)
    h_index: Optional[int] = Field(alias="hIndex", default=None)
    citations: Optional[int] = Field(default=None)
    
    model_config = ConfigDict(
        populate_by_name=True,
        # Serialize using field names (snake_case) not aliases
        by_alias=False
    )



